### PR TITLE
Add type hints for mmcv/runer/hooks/logger

### PR DESCRIPTION
**Pull request by triple-Mu**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Add type hints for mmcv/runer/hooks/logger/*

## Modification

- Update mmcv/runer/hooks/logger

## BC-breaking (Optional)

Does the modification introduce changes that break the backward-compatibility of the downstream repositories?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

**Before PR**:

- [ ] I have read and followed the workflow indicated in the [CONTRIBUTING.md](https://github.com/open-mmlab/mmcv/blob/master/CONTRIBUTING.md) to create this PR.
- [ ] Pre-commit or linting tools indicated in [CONTRIBUTING.md](https://github.com/open-mmlab/mmcv/blob/master/CONTRIBUTING.md) are used to fix the potential lint issues.
- [ ] Bug fixes are covered by unit tests, the case that causes the bug should be added in the unit tests.
- [ ] New functionalities are covered by complete unit tests. If not, please add more unit test to ensure the correctness.
- [ ] The documentation has been modified accordingly, including docstring or example tutorials.

**After PR**:

- [ ] If the modification has potential influence on downstream or other related projects, this PR should be tested with some of those projects, like MMDet or MMCls.
- [ ] CLA has been signed and all committers have signed the CLA in this PR.

### CODE REVIEW

**Review by HAOCHENYE — COMMENTED** [review]
`lr_updater` conflict with https://github.com/open-mmlab/mmcv/pull/2002/files#diff-c37a3df332ff01095b1af1a873c0ce5afe6d8289756cb5d32b036783d6ee703c

File: mmcv/runner/hooks/logger/base.py [hunk 1]
```diff
@@ -23,21 +24,23 @@ class LoggerHook(Hook):
...
+                 reset_flag: bool = False,
+                 by_epoch: bool = True):
         self.interval = interval
         self.ignore_last = ignore_last
         self.reset_flag = reset_flag
         self.by_epoch = by_epoch
 
     @abstractmethod
-    def log(self, runner):
+    def log(self, runner: Any):
```
Comment (HAOCHENYE) [inline mmcv/runner/hooks/logger/base.py:37]: We don't need to add type hints for runner in the hook (avoid circular import).

Comment (zhouzaida) [inline mmcv/runner/hooks/logger/base.py:37]:
```suggestion
    def log(self, runner):
```

File: mmcv/runner/hooks/logger/base.py [hunk 2]
```diff
@@ -57,7 +60,7 @@ def is_scalar(val, include_np=True, include_torch=True):
         else:
             return False
 
-    def get_mode(self, runner):
+    def get_mode(self, runner: Any) -> str:
```
Comment (zhouzaida) [inline mmcv/runner/hooks/logger/base.py:63]:
```suggestion
    def get_mode(self, runner) -> str:
```

File: mmcv/runner/hooks/logger/base.py [hunk 3]
```diff
@@ -70,7 +73,7 @@ def get_mode(self, runner):
                              f'but got {runner.mode}')
         return mode
 
-    def get_epoch(self, runner):
+    def get_epoch(self, runner: Any) -> int:
```
Comment (zhouzaida) [inline mmcv/runner/hooks/logger/base.py:76]:
```suggestion
    def get_epoch(self, runner) -> int:
```

File: mmcv/runner/hooks/logger/base.py [hunk 4]
```diff
@@ -82,15 +85,15 @@ def get_epoch(self, runner):
                              f'but got {runner.mode}')
         return epoch
 
-    def get_iter(self, runner, inner_iter=False):
+    def get_iter(self, runner: Any, inner_iter: bool = False) -> int:
```
Comment (zhouzaida) [inline mmcv/runner/hooks/logger/base.py:88]:
```suggestion
    def get_iter(self, runner, inner_iter: bool = False) -> int:
```

File: mmcv/runner/hooks/logger/base.py [hunk 5]
```diff
@@ -82,15 +85,15 @@ def get_epoch(self, runner):
...
         """Get the current training iteration step."""
         if self.by_epoch and inner_iter:
             current_iter = runner.inner_iter + 1
         else:
             current_iter = runner.iter + 1
         return current_iter
 
-    def get_lr_tags(self, runner):
+    def get_lr_tags(self, runner: Any) -> Dict:
```
Comment (zhouzaida) [inline mmcv/runner/hooks/logger/base.py:96]:
```suggestion
    def get_lr_tags(self, runner) -> Dict[str, float]:
```

File: mmcv/runner/hooks/logger/base.py [hunk 6]
```diff
@@ -100,7 +103,7 @@ def get_lr_tags(self, runner):
             tags['learning_rate'] = lrs[0]
         return tags
 
-    def get_momentum_tags(self, runner):
+    def get_momentum_tags(self, runner: Any) -> Dict:
```
Comment (zhouzaida) [inline mmcv/runner/hooks/logger/base.py:106]:
```suggestion
    def get_momentum_tags(self, runner) -> Dict[str, float]:
```

File: mmcv/runner/hooks/logger/base.py [hunk 7]
```diff
@@ -110,12 +113,14 @@ def get_momentum_tags(self, runner):
...
 
-    def get_loggable_tags(self,
-                          runner,
-                          allow_scalar=True,
-                          allow_text=False,
-                          add_mode=True,
-                          tags_to_skip=('time', 'data_time')):
+    def get_loggable_tags(
+        self,
+        runner: Any,
```
Comment (zhouzaida) [inline mmcv/runner/hooks/logger/base.py:118]:
```suggestion
        runner,
```

File: mmcv/runner/hooks/logger/base.py [hunk 8]
```diff
@@ -131,16 +136,16 @@ def get_loggable_tags(self,
         tags.update(self.get_momentum_tags(runner))
         return tags
 
-    def before_run(self, runner):
+    def before_run(self, runner: Any) -> None:
```
Comment (zhouzaida) [inline mmcv/runner/hooks/logger/base.py:139]:
```suggestion
    def before_run(self, runner) -> None:
```

File: mmcv/runner/hooks/logger/base.py [hunk 9]
```diff
@@ -131,16 +136,16 @@ def get_loggable_tags(self,
...
         for hook in runner.hooks[::-1]:
             if isinstance(hook, LoggerHook):
                 hook.reset_flag = True
                 break
 
-    def before_epoch(self, runner):
+    def before_epoch(self, runner: Any) -> None:
```
Comment (zhouzaida) [inline mmcv/runner/hooks/logger/base.py:145]:
```suggestion
    def before_epoch(self, runner) -> None:
```

File: mmcv/runner/hooks/logger/base.py [hunk 10]
```diff
@@ -131,16 +136,16 @@ def get_loggable_tags(self,
...
         runner.log_buffer.clear()  # clear logs of last epoch
 
-    def after_train_iter(self, runner):
+    def after_train_iter(self, runner: Any) -> None:
```
Comment (zhouzaida) [inline mmcv/runner/hooks/logger/base.py:148]:
```suggestion
    def after_train_iter(self, runner) -> None:
```

File: mmcv/runner/hooks/logger/base.py [hunk 11]
```diff
@@ -154,13 +159,13 @@ def after_train_iter(self, runner):
             if self.reset_flag:
                 runner.log_buffer.clear_output()
 
-    def after_train_epoch(self, runner):
+    def after_train_epoch(self, runner: Any) -> None:
```
Comment (zhouzaida) [inline mmcv/runner/hooks/logger/base.py:162]:
```suggestion
    def after_train_epoch(self, runner) -> None:
```

File: mmcv/runner/hooks/logger/base.py [hunk 12]
```diff
@@ -154,13 +159,13 @@ def after_train_iter(self, runner):
...
         if runner.log_buffer.ready:
             self.log(runner)
             if self.reset_flag:
                 runner.log_buffer.clear_output()
 
-    def after_val_epoch(self, runner):
+    def after_val_epoch(self, runner: Any) -> None:
```
Comment (zhouzaida) [inline mmcv/runner/hooks/logger/base.py:168]:
```suggestion
    def after_val_epoch(self, runner) -> None:
```

File: mmcv/runner/hooks/logger/clearml.py [hunk 13]
```diff
@@ -47,14 +49,14 @@ def import_clearml(self):
         self.clearml = clearml
 
     @master_only
-    def before_run(self, runner):
+    def before_run(self, runner: Any) -> None:
```
Comment (zhouzaida) [inline mmcv/runner/hooks/logger/clearml.py:52]:
```suggestion
    def before_run(self, runner) -> None:
```

File: mmcv/runner/hooks/logger/clearml.py [hunk 14]
```diff
@@ -47,14 +49,14 @@ def import_clearml(self):
...
         super().before_run(runner)
         task_kwargs = self.init_kwargs if self.init_kwargs else {}
         self.task = self.clearml.Task.init(**task_kwargs)
         self.task_logger = self.task.get_logger()
 
     @master_only
-    def log(self, runner):
+    def log(self, runner: Any) -> None:
```
Comment (zhouzaida) [inline mmcv/runner/hooks/logger/clearml.py:59]:
```suggestion
    def log(self, runner) -> None:
```

File: mmcv/runner/hooks/logger/dvclive.py [hunk 15]
```diff
@@ -50,15 +51,15 @@ def import_dvclive(self, **kwargs):
         self.dvclive = Live(**kwargs)
 
     @master_only
-    def log(self, runner):
+    def log(self, runner: Any) -> None:
```
Comment (zhouzaida) [inline mmcv/runner/hooks/logger/dvclive.py:54]:
```suggestion
    def log(self, runner) -> None:
```

File: mmcv/runner/hooks/logger/dvclive.py [hunk 16]
```diff
@@ -50,15 +51,15 @@ def import_dvclive(self, **kwargs):
...
         tags = self.get_loggable_tags(runner)
         if tags:
             self.dvclive.set_step(self.get_iter(runner))
             for k, v in tags.items():
                 self.dvclive.log(k, v)
 
     @master_only
-    def after_train_epoch(self, runner):
+    def after_train_epoch(self, runner: Any) -> None:
```
Comment (zhouzaida) [inline mmcv/runner/hooks/logger/dvclive.py:62]:
```suggestion
    def after_train_epoch(self, runner) -> None:
```

File: mmcv/runner/hooks/logger/mlflow.py [hunk 17]
```diff
@@ -57,21 +59,21 @@ def import_mlflow(self):
         self.mlflow_pytorch = mlflow_pytorch
 
     @master_only
-    def before_run(self, runner):
+    def before_run(self, runner: Any) -> None:
```
Comment (zhouzaida) [inline mmcv/runner/hooks/logger/mlflow.py:62]:
```suggestion
    def before_run(self, runner) -> None:
```

File: mmcv/runner/hooks/logger/mlflow.py [hunk 18]
```diff
@@ -57,21 +59,21 @@ def import_mlflow(self):
...
         super().before_run(runner)
         if self.exp_name is not None:
             self.mlflow.set_experiment(self.exp_name)
         if self.tags is not None:
             self.mlflow.set_tags(self.tags)
 
     @master_only
-    def log(self, runner):
+    def log(self, runner: Any) -> None:
```
Comment (zhouzaida) [inline mmcv/runner/hooks/logger/mlflow.py:70]:
```suggestion
    def log(self, runner) -> None:
```

File: mmcv/runner/hooks/logger/mlflow.py [hunk 19]
```diff
@@ -57,21 +59,21 @@ def import_mlflow(self):
...
         tags = self.get_loggable_tags(runner)
         if tags:
             self.mlflow.log_metrics(tags, step=self.get_iter(runner))
 
     @master_only
-    def after_run(self, runner):
+    def after_run(self, runner: Any) -> None:
```
Comment (zhouzaida) [inline mmcv/runner/hooks/logger/mlflow.py:76]:
```suggestion
    def after_run(self, runner) -> None:
```

File: mmcv/runner/hooks/logger/neptune.py [hunk 20]
```diff
@@ -64,24 +66,24 @@ def import_neptune(self):
         self.run = None
 
     @master_only
-    def before_run(self, runner):
+    def before_run(self, runner: Any) -> None:
```
Comment (zhouzaida) [inline mmcv/runner/hooks/logger/neptune.py:69]:
```suggestion
    def before_run(self, runner) -> None:
```

File: mmcv/runner/hooks/logger/neptune.py [hunk 21]
```diff
@@ -64,24 +66,24 @@ def import_neptune(self):
...
         if self.init_kwargs:
             self.run = self.neptune.init(**self.init_kwargs)
         else:
             self.run = self.neptune.init()
 
     @master_only
-    def log(self, runner):
+    def log(self, runner: Any) -> None:
```
Comment (zhouzaida) [inline mmcv/runner/hooks/logger/neptune.py:76]:
```suggestion
    def log(self, runner) -> None:
```

File: mmcv/runner/hooks/logger/neptune.py [hunk 22]
```diff
@@ -64,24 +66,24 @@ def import_neptune(self):
...
                         tag_value, step=self.get_iter(runner))
                 else:
                     tags['global_step'] = self.get_iter(runner)
-                    self.run[tag_name].log(tags)
+                    self.run[tag_name].log(tags)  # type: ignore
 
     @master_only
-    def after_run(self, runner):
-        self.run.stop()
+    def after_run(self, runner: Any) -> None:
```
Comment (zhouzaida) [inline mmcv/runner/hooks/logger/neptune.py:88]:
```suggestion
    def after_run(self, runner) -> None:
```

File: mmcv/runner/hooks/logger/pavi.py [hunk 23]
```diff
@@ -32,22 +33,22 @@ class PaviLoggerHook(LoggerHook):
...
+                 img_key: str = 'img_info'):
         super().__init__(interval, ignore_last, reset_flag, by_epoch)
         self.init_kwargs = init_kwargs
         self.add_graph = add_graph
         self.add_last_ckpt = add_last_ckpt
         self.img_key = img_key
 
     @master_only
-    def before_run(self, runner):
+    def before_run(self, runner: Any) -> None:
```
Comment (zhouzaida) [inline mmcv/runner/hooks/logger/pavi.py:51]:
```suggestion
    def before_run(self, runner) -> None:
```

File: mmcv/runner/hooks/logger/pavi.py [hunk 24]
```diff
@@ -85,22 +86,22 @@ def before_run(self, runner):
                 self.init_kwargs['session_text'] = session_text
         self.writer = SummaryWriter(**self.init_kwargs)
 
-    def get_step(self, runner):
+    def get_step(self, runner: Any) -> int:
```
Comment (zhouzaida) [inline mmcv/runner/hooks/logger/pavi.py:89]:
```suggestion
    def get_step(self, runner) -> int:
```

File: mmcv/runner/hooks/logger/pavi.py [hunk 25]
```diff
@@ -85,22 +86,22 @@ def before_run(self, runner):
...
         """Get the total training step/epoch."""
         if self.get_mode(runner) == 'val' and self.by_epoch:
             return self.get_epoch(runner)
         else:
             return self.get_iter(runner)
 
     @master_only
-    def log(self, runner):
+    def log(self, runner: Any) -> None:
```
Comment (zhouzaida) [inline mmcv/runner/hooks/logger/pavi.py:97]:
```suggestion
    def log(self, runner) -> None:
```

File: mmcv/runner/hooks/logger/pavi.py [hunk 26]
```diff
@@ -85,22 +86,22 @@ def before_run(self, runner):
...
         tags = self.get_loggable_tags(runner, add_mode=False)
         if tags:
             self.writer.add_scalars(
                 self.get_mode(runner), tags, self.get_step(runner))
 
     @master_only
-    def after_run(self, runner):
+    def after_run(self, runner: Any) -> None:
```
Comment (zhouzaida) [inline mmcv/runner/hooks/logger/pavi.py:104]:
```suggestion
    def after_run(self, runner) -> None:
```

File: mmcv/runner/hooks/logger/pavi.py [hunk 27]
```diff
@@ -118,7 +119,7 @@ def after_run(self, runner):
         self.writer.close()
 
     @master_only
-    def before_epoch(self, runner):
+    def before_epoch(self, runner: Any) -> None:
```
Comment (zhouzaida) [inline mmcv/runner/hooks/logger/pavi.py:122]:
```suggestion
    def before_epoch(self, runner) -> None:
```

File: mmcv/runner/hooks/logger/segmind.py [hunk 28]
```diff
@@ -40,7 +42,7 @@ def import_segmind(self):
         self.mlflow_log = segmind.utils.logging_utils.try_mlflow_log
 
     @master_only
-    def log(self, runner):
+    def log(self, runner: Any) -> None:
```
Comment (zhouzaida) [inline mmcv/runner/hooks/logger/segmind.py:45]:
```suggestion
    def log(self, runner) -> None:
```

**Comment by zhouzaida** [comment]
Please add type hints for this [method](https://github.com/open-mmlab/mmcv/pull/2000/files#diff-97591557165ae12b1eeed0a5fc08d1e8cfd6b8d01abcb5a02eb3066e1849ee28L33)

File: mmcv/runner/hooks/logger/tensorboard.py [hunk 29]
```diff
@@ -23,16 +24,16 @@ class TensorboardLoggerHook(LoggerHook):
...
+                 interval: int = 10,
+                 ignore_last: bool = True,
+                 reset_flag: bool = False,
+                 by_epoch: bool = True):
         super().__init__(interval, ignore_last, reset_flag, by_epoch)
         self.log_dir = log_dir
 
     @master_only
-    def before_run(self, runner):
+    def before_run(self, runner: Any) -> None:
```
Comment (zhouzaida) [inline mmcv/runner/hooks/logger/tensorboard.py:36]:
```suggestion
    def before_run(self, runner) -> None:
```

File: mmcv/runner/hooks/logger/tensorboard.py [hunk 30]
```diff
@@ -55,7 +56,7 @@ def before_run(self, runner):
         self.writer = SummaryWriter(self.log_dir)
 
     @master_only
-    def log(self, runner):
+    def log(self, runner: Any) -> None:
```
Comment (zhouzaida) [inline mmcv/runner/hooks/logger/tensorboard.py:59]:
```suggestion
    def log(self, runner) -> None:
```

File: mmcv/runner/hooks/logger/tensorboard.py [hunk 31]
```diff
@@ -64,5 +65,5 @@ def log(self, runner):
                 self.writer.add_scalar(tag, val, self.get_iter(runner))
 
     @master_only
-    def after_run(self, runner):
+    def after_run(self, runner: Any) -> None:
```
Comment (zhouzaida) [inline mmcv/runner/hooks/logger/tensorboard.py:68]:
```suggestion
    def after_run(self, runner) -> None:
```

File: mmcv/runner/hooks/logger/text.py [hunk 32]
```diff
@@ -2,7 +2,7 @@
 import datetime
 import os
 import os.path as osp
-from collections import OrderedDict
+from typing import Any, Dict, List, Optional, OrderedDict, Union
```
Comment (zhouzaida) [inline mmcv/runner/hooks/logger/text.py:5]: OrderedDict can not be imported from typing when python<=3.6

File: mmcv/runner/hooks/logger/text.py [hunk 33]
```diff
@@ -53,15 +53,15 @@ class TextLoggerHook(LoggerHook):
...
-                 out_suffix=('.log.json', '.log', '.py'),
-                 keep_local=True,
-                 file_client_args=None):
+                 by_epoch: bool = True,
+                 interval: int = 10,
+                 ignore_last: bool = True,
+                 reset_flag: bool = False,
+                 interval_exp_name: int = 1000,
+                 out_dir: Optional[str] = None,
+                 out_suffix: tuple = ('.log.json', '.log', '.py'),
```
Comment (zhouzaida) [inline mmcv/runner/hooks/logger/text.py:62]:
```suggestion
                 out_suffix: Union[str, tuple] = ('.log.json', '.log', '.py'),
```

File: mmcv/runner/hooks/logger/text.py [hunk 34]
```diff
@@ -85,7 +85,7 @@ def __init__(self,
             self.file_client = FileClient.infer_client(file_client_args,
                                                        self.out_dir)
 
-    def before_run(self, runner):
+    def before_run(self, runner: Any) -> None:
```
Comment (zhouzaida) [inline mmcv/runner/hooks/logger/text.py:88]:
```suggestion
    def before_run(self, runner) -> None:
```

File: mmcv/runner/hooks/logger/text.py [hunk 35]
```diff
@@ -105,7 +105,7 @@ def before_run(self, runner):
         if runner.meta is not None:
             self._dump_log(runner.meta, runner)
 
-    def _get_max_memory(self, runner):
+    def _get_max_memory(self, runner: Any) -> int:
```
Comment (zhouzaida) [inline mmcv/runner/hooks/logger/text.py:108]:
```suggestion
    def _get_max_memory(self, runner) -> int:
```

File: mmcv/runner/hooks/logger/text.py [hunk 36]
```diff
@@ -115,7 +115,7 @@ def _get_max_memory(self, runner):
             dist.reduce(mem_mb, 0, op=dist.ReduceOp.MAX)
         return mem_mb.item()
 
-    def _log_info(self, log_dict, runner):
+    def _log_info(self, log_dict: Dict, runner: Any) -> None:
```
Comment (zhouzaida) [inline mmcv/runner/hooks/logger/text.py:118]:
```suggestion
    def _log_info(self, log_dict: Dict, runner) -> None:
```

File: mmcv/runner/hooks/logger/text.py [hunk 37]
```diff
@@ -181,7 +181,7 @@ def _log_info(self, log_dict, runner):
 
         runner.logger.info(log_str)
 
-    def _dump_log(self, log_dict, runner):
+    def _dump_log(self, log_dict: Dict, runner: Any) -> None:
```
Comment (zhouzaida) [inline mmcv/runner/hooks/logger/text.py:184]:
```suggestion
    def _dump_log(self, log_dict: Dict, runner) -> None:
```

File: mmcv/runner/hooks/logger/text.py [hunk 38]
```diff
@@ -192,15 +192,15 @@ def _dump_log(self, log_dict, runner):
                 mmcv.dump(json_log, f, file_format='json')
                 f.write('\n')
 
-    def _round_float(self, items):
+    def _round_float(self, items: Union[List, float]):
```
Comment (zhouzaida) [inline mmcv/runner/hooks/logger/text.py:195]: items may have other types so it does need type hints here.

File: mmcv/runner/hooks/logger/text.py [hunk 39]
```diff
@@ -192,15 +192,15 @@ def _dump_log(self, log_dict, runner):
...
         if isinstance(items, list):
             return [self._round_float(item) for item in items]
         elif isinstance(items, float):
             return round(items, 5)
         else:
             return items
 
-    def log(self, runner):
+    def log(self, runner: Any) -> OrderedDict:
```
Comment (zhouzaida) [inline mmcv/runner/hooks/logger/text.py:203]:
```suggestion
    def log(self, runner) -> OrderedDict:
```

File: mmcv/runner/hooks/logger/text.py [hunk 40]
```diff
@@ -228,13 +228,13 @@ def log(self, runner):
...
 
-        log_dict = dict(log_dict, **runner.log_buffer.output)
+        log_dict = dict(log_dict, **runner.log_buffer.output)  # type: ignore
 
         self._log_info(log_dict, runner)
         self._dump_log(log_dict, runner)
         return log_dict
 
-    def after_run(self, runner):
+    def after_run(self, runner: Any) -> None:
```
Comment (zhouzaida) [inline mmcv/runner/hooks/logger/text.py:237]:
```suggestion
    def after_run(self, runner) -> None:
```

File: mmcv/runner/hooks/logger/wandb.py [hunk 41]
```diff
@@ -48,15 +49,15 @@ class WandbLoggerHook(LoggerHook):
...
-                 out_suffix=('.log.json', '.log', '.py')):
+                 init_kwargs: Optional[Dict] = None,
+                 interval: int = 10,
+                 ignore_last: bool = True,
+                 reset_flag: bool = False,
+                 commit: bool = True,
+                 by_epoch: bool = True,
+                 with_step: bool = True,
+                 log_artifact: bool = True,
+                 out_suffix: tuple = ('.log.json', '.log', '.py')):
```
Comment (zhouzaida) [inline mmcv/runner/hooks/logger/wandb.py:60]:
```suggestion
                 out_suffix: Union[str, tuple] = ('.log.json', '.log', '.py')):
```

File: mmcv/runner/hooks/logger/wandb.py [hunk 42]
```diff
@@ -74,17 +75,17 @@ def import_wandb(self):
         self.wandb = wandb
 
     @master_only
-    def before_run(self, runner):
+    def before_run(self, runner: Any) -> None:
```
Comment (zhouzaida) [inline mmcv/runner/hooks/logger/wandb.py:78]:
```suggestion
    def before_run(self, runner) -> None:
```

File: mmcv/runner/hooks/logger/wandb.py [hunk 43]
```diff
@@ -74,17 +75,17 @@ def import_wandb(self):
...
         if self.init_kwargs:
-            self.wandb.init(**self.init_kwargs)
+            self.wandb.init(**self.init_kwargs)  # type: ignore
         else:
-            self.wandb.init()
+            self.wandb.init()  # type: ignore
 
     @master_only
-    def log(self, runner):
+    def log(self, runner: Any) -> None:
```
Comment (zhouzaida) [inline mmcv/runner/hooks/logger/wandb.py:88]:
```suggestion
    def log(self, runner) -> None:
```

File: mmcv/runner/hooks/logger/wandb.py [hunk 44]
```diff
@@ -95,7 +96,7 @@ def log(self, runner):
                 self.wandb.log(tags, commit=self.commit)
 
     @master_only
-    def after_run(self, runner):
+    def after_run(self, runner: Any) -> None:
```
Comment (zhouzaida) [inline mmcv/runner/hooks/logger/wandb.py:99]:
```suggestion
    def after_run(self, runner) -> None:
```

**Review by zhouzaida — APPROVED** [review]

**Pull request merged**